src/components/AnswerOption.tsx

```diff
@@ -0,0 +1,38 @@
+import React from "react";
+
+type Props = {
+  answer: string;
+  index: number;
+  isCorrect: boolean;
+  isSelected: boolean;
+  isSubmitted: boolean;
+  stateHook: (value: React.SetStateAction<number | null>) => void;
+};
+
+export const AnswerOption: React.FC<Props> = ({
+  answer,
+  index,
+  isCorrect,
+  isSelected,
+  isSubmitted,
+  stateHook,
+}: Props) => {
+  let buttonClass: string = "answer-btn";
+
+  if (isSubmitted && isCorrect) buttonClass += " correct";
+
+  if (isSubmitted && isSelected && !isCorrect) buttonClass += " incorrect";
+
+  if (!isSubmitted && isSelected) buttonClass += " selected";
+
+  return (
+    <button
+      key={index}
```

Comment:
We can remove this key (and the index prop) as we take care of the key properly in the parent component :)